config/config.go

```diff
@@ -686,11 +686,13 @@ func (c *Config) LoadDirectory(path string) error {
 }
 
 // Try to find a default config file at these locations (in order):
-//   1. $TELEGRAF_CONFIG_PATH
-//   2. $HOME/.telegraf/telegraf.conf
-//   3. /etc/telegraf/telegraf.conf
+//   1. $TELEGRAF_CONFIG_URL
+//   2. $TELEGRAF_CONFIG_PATH
+//   3. $HOME/.telegraf/telegraf.conf
+//   4. /etc/telegraf/telegraf.conf
 //
 func getDefaultConfigPath() (string, error) {
+	urlfile := os.Getenv("TELEGRAF_CONFIG_URL")
 	envfile := os.Getenv("TELEGRAF_CONFIG_PATH")
```

Comment:
The -config command line option takes either a local path or a URL.  I think we should also have one environment variable that takes either a local path or a URL.  I would prefer to keep TELEGRAF_CONFIG_PATH for backward compatibility and allow it to accept URLs.

It is unfortunate that the "_PATH" might make users think it only accepts local paths, but I think this is better than having two environment variables.  We should make sure to add to the documentation to let users know URLs work too.

In the long term we may want to consider renaming the single env var to something more generic like "TELEGRAF_CONFIG" but that would be a breaking change so it would have to wait for a major release (2.0.0) where compatibility can be broken.

For this PR to be merged, could you remove the new TELEGRAF_CONFIG_URL env var?  It looks like we may also need to combine the getDefaultConfigPath and LoadConfig and loadConfig functions because getDefaultConfigPath checks if the string is a file path, but loadConfig checks if it's a URL.  I think we will need to do that in the same function.

Reply:
@Kuzyashin What do you think of this approach?  Are you able to work on this PR more?

Reply:
Yes. I can push new commits in 2-3 days.

Reply:
Let me know if you can't find the time for this and we'll pass it on to someone else.

Reply:
@reimda can you please check again, I think your comment was resolved with the latest commits.